packages/functional-tests/tests/react-conversion/signup.spec.ts

```diff
@@ -72,12 +72,14 @@ test.describe('severity-1 #smoke', () => {
       // wait for navigation, and get search params
       await page.waitForURL(/oauth\//);
       const url = page.url();
-      const params = new URLSearchParams(url.substring(url.indexOf('?') + 1));
+      const params = new URLSearchParams(url);
```

Comment:
Is this... right?
Doesn't the [`URLSearchParams()` constructor](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) take just the query string and not the entire URL?

https://runkit.com/pdehaan/65667faa43112e0008913616

Reply:
You could just wrap `page.url()` in a URL constructor and then use `searchParams` property to get the underlying `searchParams` object:

```js
import { firefox } from "playwright";

const URLS = [
  "https://google.com?foo=bar&baz=boop",
  "https://bbc.com?dnt=1",
];

const browser = await firefox.launch();
const page = await browser.newPage();
for (const currentURL of URLS) {
  await page.goto(currentURL);
  const params = new URL(page.url()).searchParams;
  console.log(page.url(), params);
}
await browser.close();

// https://www.google.com/?foo=bar&baz=boop URLSearchParams { 'foo' => 'bar', 'baz' => 'boop' }
// https://www.bbc.com/?dnt=1 URLSearchParams { 'dnt' => '1' }
```

Reply:
🤦🏽‍♀️ You're absolutely right. Thank you @pdehaan.